src/python/pants/option/global_options.py

```diff
@@ -1748,6 +1748,44 @@ class BootstrapOptions:
             """
         ),
     )
+    _downloads_intrinsic_retry_delay = FloatOption(
+        default=0.2,
+        advanced=True,
+        help=softwrap(
+            """
+            Set to the duration to use as the "period" for exponential backoff between download attempts
+            by the downloads intrinsinc rule when it encounters retryable errors.
```

Comment:
If I'm a user without Pants experience, I reckon this text will feel pretty opaque. Maybe we could frame it more around how it is exposed to users, like:

```suggestion
            for downloading files via `http_source` (and any other uses of the `download_file` rule) when it encounters retryable errors.
```

And applied to the max attempts options too.  

Also, the option names changed to `file_download_retry_delay` and `file_download_max_attempts`, or similar. (Plus updated in release notes.)

(Sorry for the review dribble! I could've flagged this in the first pass.)

Reply:
Good point. Most users won't know (or care for that matter) about the fact that these options control an intrinsic rule.

Reply:
I renamed the options as suggested and made the docs hopefully easier to follow. Please let me know if I succeeded. :)